### ISSUE TITLE
fparser2: handle namelist specifiers in formatted I/O calls

### ISSUE THREAD

**Issue by arporter** [stfc/fparser#34]
When parsing a READ that uses a namelist, e.g.:

    NAMELIST/nammpp/ cn_mpi_send, nn_buffer, jpni, jpnj, jpnij, ln_nnogather

    READ  ( kumnam_ref, nammpp, IOSTAT = ios, ERR = 901)
    901   IF( ios /= 0 ) [report an error]

fparser2 incorrectly identifies the namelist specifier (`nammpp` in this example) as being the format specifier. For this example this results in it subsequently generating:

    READ(UNIT = kumnam_ref, FMT = nammpp, IOSTAT = ios, ERR = 901)

Which is incorrect. It should be:

    READ(UNIT = kumnam_ref, NML = nammpp, IOSTAT = ios, ERR = 901)

**Comment by arporter** [comment]
Created the `namelist_arg` branch for this work.

**Comment by arporter** [comment]
The relevant part of the standard is:

C917   (R913) If format appears without a preceding FMT=, it shall be the second item in the
io-control-spec-list and the first item shall be io-unit.

C918   (R913) If namelist-group-name appears without a preceding NML=, it shall be the second item
in the io-control-spec-list and the first item shall be io-unit.

Therefore, without identifying the type of the second argument, it is impossible to tell whether it is a namelist or a format specifier.

**Comment by arporter** [comment]
Identifying the type of the argument could potentially be complex (e.g. it could be defined in another Fortran module). I think the simplest solution is therefore to alter fparser so that it does not automatically *add* "FMT=" if it isn't present in the original code.

**Comment by arporter** [comment]
Discovered that `Io_Control_Spec` uses `KeywordValueBase.match` and this can be told whether or not there must be a LHS to the expression it is matching. I've therefore set this to False for Unit, Format and Namelist. Main change is to `Io_Control_Spec_List.match()` though - it now only adds "FMT=" if the second argument is positional and either a character string constant or a "*". Otherwise we don't know whether we've got a namelist or a format so we don't prepend anything.

Had to correct some of the related tests. I've taken the opportunity to tidy for pep8/pylint in the code that I had to change/examine.
